From 9d71b61c5ca8222fba170fd5f7c5be18e0cde322 Mon Sep 17 00:00:00 2001
From: Olliver Schinagl <oliver@schinagl.nl>
Date: Tue, 19 Jan 2021 20:32:07 +0100
Subject: [PATCH] io: Handle really long lines

While commit 18b0b45 (io: Handle long lines, Thu Jan 7 17:25:23 2021
+0100) did attempt to address this issue, the buffer really is still to
small when dealing with big-big dependency lists.

Lets make it sufficiently large for now, until the new APKINDEX format
can support multi-line dependencies, making this not needed any more.

[TT: Originally the buffer size was conservative to run on resource
constrained embedded platforms. But since the available memory on those
has also increased much, the adjustment to 128kB makes sense also to
increase performance a little bit. Backported to 2.10-stable.]

Signed-off-by: Olliver Schinagl <oliver@schinagl.nl>
---
 src/io.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/io.c b/src/io.c
index 47f34bf..414e4e2 100644
--- a/src/io.c
+++ b/src/io.c
@@ -213,7 +213,7 @@ struct apk_istream_bstream {
 	struct apk_bstream bs;
 	struct apk_istream *is;
 	apk_blob_t left;
-	char buffer[8*1024];
+	char buffer[128*1024];
 };
 
 static void is_bs_get_meta(struct apk_bstream *bs, struct apk_file_meta *meta)
-- 
2.35.1

